docs/cvl/invariants.md

```diff
@@ -55,16 +55,19 @@ production, and {doc}`statements` for the `block` production.
 Overview
 --------
 
-In CVL, an invariant is a property of the contract state that is expected to be
+In CVL, we distinguish between strong and weak invariants.
+A *weak* invariant is a property that is expected to be
 true whenever a contract method is not currently executing.  This kind of
-invariant is sometimes called a "representation invariant".
+invariant is sometimes called a "representation invariant". A *strong* invariant is 
+an invariant that also holds before and after execution of an unresolved call, i.e. a call that
+potentially calls to another contract which could modify the current contract's state. 
```

Comment:
This paragraph still doesn't explain when should we use strong invariants over weak invariants. Perhaps we should base on the first sentence from the example at https://github.com/Certora/Examples/blob/cli-beta/CVLByExample/StrongInvariants/README.md:

"Strong invariants are sure to hold whenever control is yielded to an external function, providing enhanced security, especially for contracts without global locks. However, they add additional checks that can result in longer run times or even halted jobs."

Reply:
In general, linking to the example would be highly beneficial

Reply:
Thanks - I linked the example and improved the wording.